presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java

```diff
@@ -1289,6 +1289,22 @@ public void testWidthBucketArray()
         assertFunction("width_bucket(1.5E0, array[1.0E0, 2.3E0, 2.0E0])", BIGINT, 1L);
     }
 
+    @Test
+    public void testWidthBucketNamesArray()
+    {
+        assertFunction("width_bucket(3.14E0, array[0.0E0, 2.0E0, 4.0E0], array['a', 'b', 'c', 'd'])", VarcharType.VARCHAR, "c");
```

Comment:
@amitkdutta Looks like you can achieve what you want with the existing functions, e.g. `select array['a', 'b', 'c', 'd'][width_bucket(3.14E0, array[0.0E0, 2.0E0, 4.0E0]) + 1]`

Reply:
Its possible to write that way, however in Cubrick we supported the syntax with a bucket name. We are shadowing traffic from Cubrick to Presto, and thought this syntax can be useful. Re-writing the query the you mentioned is a bit non-trivial, while adding another parameter seemed ok to me.

Reply:
In Presto, we try to limit the number of variations of a function. Imagine someone else would like to add another variation that takes a map of buckets and names. Then another user would like to add a variation that takes an array of structs (value + name)/. There can be a very large number of variations that are convenient in different cases.